dotnet/src/SemanticKernel.Abstractions/Search/KernelSearchResults.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.SemanticKernel.Search;
+
+/// <summary>
+/// Represents the search results returned from a <see cref="ITextSearch" /> service.
+/// </summary>
+[Experimental("SKEXP0001")]
+public class KernelSearchResults<T>
+{
+    /// <summary>
+    /// The total count of results found by the search operation, or null
+    /// if the count was not requested.
```

Comment:
I don't see an option for requesting the total count.
Also, is this the total count regardless of how many were returned in the first page, e.g. someone asked for Count = 3, but there are 20 matches in the store.  Should this be 3 or 20.  The latter may not be available everywhere I expect.